src/utils/file-upload.ts

```diff
@@ -9,11 +9,78 @@ export interface OutputFile {
   }
 }
 
+/**
+ * Interface describing a presigned post response
+ */
+export interface PresignedPostResponse {
+  error: boolean,
+  data: PresignedPostData,
+  message: string | null
+}
+
+/**
+ * Interface describing a presigned post data
+ */
+export interface PresignedPostData {
+  url: string,
+  fields: any
+}
+
 /**
  * Utility class for uploading files
  */
 export default class FileUpload {
 
+  /**
+   * Retrieve pre-signed POST data from a dedicated API endpoint.
+   *
+   * @param folder folder
+   * @param file file
+   * @returns {Promise<any>}
+   */
+  public static getPresignedPostData = (folder: string, file: File): Promise<PresignedPostResponse> => {
+    return new Promise(resolve => {
+      const xhr = new XMLHttpRequest();
+      const url = process.env.REACT_APP_USER_CONTENT_UPLOAD_URL as string;
+
+      xhr.open("POST", url, true);
+      xhr.setRequestHeader("Content-Type", "application/json");
+      xhr.send(
+        JSON.stringify({
+          path: `${folder}/${file.name}`,
+          type: file.type
+        })
+      );
+      xhr.onload = function() {
+        resolve(JSON.parse(this.responseText));
+      };
+    });
+  }
+
+  /**
+   * Upload file to S3 with previously received pre-signed POST data.
+   *
+   * @param presignedPostData
+   * @param file
+   * @returns {Promise<any>}
```

Comment:
Define return type and replace any with proper type or remove this if there is no return type